llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -957,6 +986,14 @@ static Value *tryToFoldLiveIns(const VPRecipeBase &R, unsigned Opcode,
   }
 
   InstSimplifyFolder Folder(DL);
+  if (OpcodeOrIID->first) {
+    if (R.getNumOperands() != 2)
+      return nullptr;
+    unsigned ID = OpcodeOrIID->second;
+    return Folder.FoldBinaryIntrinsic(
+        ID, Ops[0], Ops[1], TypeInfo.inferScalarType(R.getVPSingleValue()));
```

Comment:
```suggestion
        ID, Ops[0], Ops[1], TypeInfo.inferScalarType(R));
```

passed now was VPSingleDefRecipe